lib/ui/common/roundedButton.dart

```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
+  const RoundedButton({Key key, RoundedButtonViewModel viewModel, RoundedButtonStyle style }) : this._viewModel = viewModel, this._style = style, super(key: key);
+
+  static Widget _build(RoundedButtonViewModel viewModel, RoundedButtonStyle style) {
 
     List<Widget> _buildButtonContent(){
       List<Widget> listBuilder = [];
       if (viewModel.icon != null) {
         listBuilder.add(
-            Icon(
-              viewModel.icon,
-              size: buttonStyle.buttonIconSize,
-              color: buttonStyle.iconButtonColor,
-            )
+            Image.asset(viewModel.icon, height: style.buttonIconSize)
         );
       }
       if (viewModel.title != null) {
         listBuilder.add(
             Text(
                 viewModel.title,
-                style: buttonStyle.buttonTextStyle
+                style: style.buttonTextStyle
             ));
       }
       return listBuilder;
     }
 
     return GestureDetector(
-      onTap: () => _showToast(viewModel.context),
-      child: Padding(
-        padding: buttonStyle.edgePadding,
-        child: Container(
-          alignment: Alignment.center,
-          height: buttonStyle.size,
-          width: buttonStyle.size,
-          decoration: BoxDecoration(
-              color: buttonStyle.backgroundColor,
-              shape: buttonStyle.buttonShape,
-              borderRadius: buttonStyle.borderRadius
-          ),
-          child: Wrap(
-            direction: Axis.horizontal,
-            children: _buildButtonContent(),
+      onTap: () => viewModel.onTap(),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: <Widget>[
+          Padding(
+            padding: style.edgePadding,
+            child: Container(
+              alignment: Alignment.center,
+              height: style.height,
+              width: style.width,
+              decoration: BoxDecoration(
+                  color: viewModel.backgroundColor != null ? viewModel.backgroundColor : style.backgroundColor,
+                  shape: style.buttonShape,
+                  borderRadius: style.borderRadius
+              ),
+              child: Wrap(
+                direction: Axis.horizontal,
+                children: _buildButtonContent(),
+              ),
+            ),
           ),
-        ),
+        ],
       ),
     );
   }
 
-//FIXME: Only is built for show that this buttons are not functional yet
-  static void _showToast(BuildContext context) {
-    final String toastText = "Not Implemented Yet";
-    final String toastButtonText = "BACK";
-    final scaffold = Scaffold.of(context);
-    scaffold.showSnackBar(
-        SnackBar(
-          content: Text(
-              toastText
-          ),
-          action: SnackBarAction(label: toastButtonText, onPressed: scaffold.hideCurrentSnackBar),
-        )
-    );
+  @override
+  Widget build(BuildContext context) {
+    return _build(_viewModel, _style);
```

Comment:
Move all the stuff inside _build here and remove the _build function. Remember that @override funcs should be before private and public funcs